### DISCUSSION TITLE
New inbound FBA API 2024-03-20 -- Can't use 2D Barcodes or APC when providing SKU/QTY??

### DISCUSSION THREAD

**Discussion by MetaMan2023** [amzn/selling-partner-api-models#4352] [Q&A]
We are trying to finish converting the inbound FBA API from v0 to V2024-03-20 before the deadline and have hit a new issue.

Historically we would:

(A) Provide amazon SKUs & QTY and ask what centers to go to -- Amazon would return with center splits (and what qty to go to those centers)
(B) We would pack boxes based on those splits
(C) We would then complete those specific shipments to those centers using 2D barcodes and provide box content info to ship with APC

There are two "paths" to follow in the new API -- the first path is if your boxes are packed already -- which ours are not.  The second path allows you, similar to before, provide SKU and QTY info up front and get shipment split information.  We have completed the code successfully to do this, have gotten shipping split info and "created the shipments".   Then after the FBA shipment is "created" by Amazon we went into Seller Central to complete the shipment only to find the following message:

"Step 2 is broken into three parts because you chose &quot;Skip packing and continue&quot; in Step 1b. 2a is to confirm shipments, 2b is to confirm packing details using excel sheet only, and 2c is to confirm LTL/FTL transportation only. To switch to the standard experience, you can click &quot;View/Edit&quot; on Step 1b and submit the packing details."

Basically we can do neither use 2D BARCODES nor AMAZON PARTNERED CARRIER?  

Is this correct?   It seems the only way we can use 2D barcodes or APC is to pack boxes up front which is completely backwards from the way our FBA process has worked for 15 years.

Any help please??

**Comment by puppsupr** [comment]
Hi [MetaMan2023](https://github.com/MetaMan2023),
On the right hand flow, you can't use 2D barcodes as of now. For all ship later packages, you must use BOX_CONTENT_PROVIDED to specify box dimensions, box weight, and the quantity of each box. This workflow is only limited to FTL/LTL pallets but you should be able to see Amazon Partnered carriers. If you are unable to see the partnered carriers, can you please [open a support case](https://developer.amazonservices.com/support) following the [guidance](https://developer.amazonservices.com/support-guidelines) here so we can pursue the investigation as additional details are required from your end.

**Reply by collery** [reply]
if there is any new progress here。 we use branch 2 ,and stop in step 3(setPackingInfo)

**Comment by MetaMan2023** [comment]
Just to confirm -- with right hand flow, we can send "Amazon Partnered" FTL/LTL?

Separate question:

* Is support for Amazon Partnered UPS (on right flow) realistically going to be offered?  I have read "it's not currently available" but never a timeline.   Want to know how realistic it is and if there is a specific time frame.

**Reply by puppsupr** [reply]
Yes, right hand also supports partnered carriers, requirements to get partnered carriers are as [here](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-faq#what-are-the-eligibility-dimensions-and-weight-requirements-for-spd-and-ltl-options).

**Reply by MetaMan2023** [reply]
@puppsupr 

Hi -- can you please clarify as unclear in your response.  

Are you saying that APC UPS is now available on the right flow?  Or were you just acknowledging the current status that APC FTL/LTL is available on right flow but not APC UPS?
